mava/networks/mat_network.py

```diff
@@ -0,0 +1,317 @@
+            self.log_std = None
+
+        self.obs_encoder = nn.Sequential(
+            [ln(), nn.Dense(self.n_embd, kernel_init=orthogonal(jnp.sqrt(2))), nn.gelu],
+        )
+        self.ln = ln()
+        self.blocks = [
+            DecodeBlock(
+                self.n_embd,
+                self.n_head,
+                self.n_agent,
+                use_swiglu=self.use_swiglu,
+                use_rmsnorm=self.use_swiglu,
+                name=f"cross_attention_block_{block_id}",
+            )
+            for block_id in range(self.n_block)
+        ]
+        self.head = nn.Sequential(
+            [
+                nn.Dense(self.n_embd, kernel_init=orthogonal(jnp.sqrt(2))),
+                nn.gelu,
+                ln(),
+                nn.Dense(self.action_dim, kernel_init=orthogonal(0.01)),
+            ],
+        )
+
+    def __call__(self, action: chex.Array, obs_rep: chex.Array) -> chex.Array:
+        action_embeddings = self.action_encoder(action)
+        x = self.ln(action_embeddings)
+
+        # Need to loop here because the input and output of the blocks are different.
+        # Blocks take an action embedding and observation encoding as input but only give the cross
+        # attention output as output.
+        for block in self.blocks:
+            x = block(x, obs_rep)
+        logit = self.head(x)
+
+        return logit
+
+
+class MultiAgentTransformer(nn.Module):
+    obs_dim: int
+    action_dim: int
+    n_block: int
+    n_embd: int
+    n_head: int
+    n_agent: int
+    action_space_type: str = "discrete"
+    use_swiglu: bool = False
+    use_rmsnorm: bool = False
+
+    # General shapes legend:
+    # B: batch size
+    # N: number of agents
+    # O: observation dimension
+    # A: action dimension
+    # E: model embedding dimension
+
+    def setup(self) -> None:
+        if self.action_space_type not in ["discrete", "continuous"]:
+            raise ValueError(f"Invalid action space type: {self.action_space_type}")
+
+        self.encoder = Encoder(
+            self.obs_dim,
+            self.action_dim,
+            self.n_block,
+            self.n_embd,
+            self.n_head,
+            self.n_agent,
+            use_swiglu=self.use_swiglu,
+            use_rmsnorm=self.use_rmsnorm,
+        )
+        self.decoder = Decoder(
+            self.obs_dim,
+            self.action_dim,
+            self.n_block,
+            self.n_embd,
+            self.n_head,
+            self.n_agent,
+            self.action_space_type,
+            use_swiglu=self.use_swiglu,
+            use_rmsnorm=self.use_rmsnorm,
+        )
+
+        if self.action_space_type == "discrete":
+            self.act_function = discrete_autoregressive_act
+            self.train_function = discrete_parallel_act
+        elif self.action_space_type == "continuous":
+            self.act_function = continuous_autoregressive_act
+            self.train_function = continuous_parallel_act
```

Comment:
Just need to import these from the file where we get the action space type

```suggestion
        if self.action_space_type == _DISCRETE:
            self.act_function = discrete_autoregressive_act
            self.train_function = discrete_parallel_act
        elif self.action_space_type == _CONTINUOUS:
            self.act_function = continuous_autoregressive_act
            self.train_function = continuous_parallel_act
```

Reply:
Ah yes, I forgot to do this :sweat_smile: 
Done.